scripts/schemas/twister/hwmap-schema.yaml

```diff
@@ -29,7 +29,7 @@ sequence:
         required: true
       "serial":
         type: str
-        required: false
+        required: true
```

Comment:
how does this solve the problem? you can still provide an empty string in the map file and end up with the same issue. You also need to check for valid values, solving this on the schema is not enough.

Reply:
Yes, this is not enough to solve this problem. This will works only if `(d.serial or d.serial_pty)` checking will be removed from  `device_is_available` method from `DeviceHandler` class. 
If it will be provided, that serial will be required in hardware map, than it will be unnecessary to check `(d.serial or d.serial_pty)` in `device_is_available` method from `DeviceHandler` class. Due to removing this checking, when empty string as serial port will be passed, then program will not get stuck in endless while loop in `handle` method, but go forward and during trying to create `Serial` object (with empty string as a path argument) program will rise an error.